### DISCUSSION TITLE
Retrieving values from NotificationCCReports directly ?

### DISCUSSION THREAD

**Discussion by lowlighter** [zwave-js/zwave-js#2886] [Q&A] [answered]
Hello!

Thanks a lot for this awesome library 🙂 !

I'm currently experimenting a bit with it and was wondering if there was a way to get "notification state" ("Window/door is open" in this case) which seems to be present in the NotificationCCReport but not in the event exposed through regulars listeners (I've tried `notification`, `value notification`, `value added`, `value removed`, `value updated`) ?

Values returned in this case are 22 (open)/23 (closed) but it's not very practical to handle since it forces me to map possible states for all kind of sensors.

![](https://user-images.githubusercontent.com/22963968/122641553-58843380-d106-11eb-8bfe-1ff71ba10908.png)

Currently using node-zwave-js 7.7.4 with a Z-stick Gen 5 and a Fibaro door sensor

**Comment by AlCalzone** [comment, marked as answer]
This should be coming through the `value updated` event. If you take the value ID and look up the corresponding metadata, you should get a `ValueMetadataNumeric` https://zwave-js.github.io/node-zwave-js/#/api/valueid?id=number which has a `states` record you can use to map 22/23 to the corresponding strings.

**Reply by lowlighter** [reply]
Thanks a lot, I'll try that!

**Reply by lowlighter** [reply]
Ok it does works, if anyone needs a code snippet it looks like this:
```js
node.on("value updated", function (node, event) {
  const {label, states} = node.getValueMetadata(event)
  const {newValue, prevValue} = event
  console.log(`${label}: changed from "${states[prevValue]}" to "${states[newValue]}"`)
  //Door state: changed from "Window/door is closed" to "Window/door is open"
})
```
